src/backend/core/models.py

```diff
@@ -770,19 +770,81 @@ def get_logo_url(self):
             return f"{settings.API_PUBLIC_URL}servicelogo/{self.id}/"
         return None
 
-    def can_activate(self, organization: Organization):
+    def can_activate(self, organization: Organization, operator: "Operator" = None):
         """
         Check if the service can be activated for the given organization.
+
+        Returns:
+            tuple: (can_activate: bool, reason: str | None)
         """
-        if not self.required_services.count():
-            return True
-        return (
-            self.required_services.all()
-            .filter(
-                subscriptions__organization=organization, subscriptions__is_active=True
+        # Check required services first
+        if self.required_services.count():
+            has_required = (
+                self.required_services.all()
+                .filter(
+                    subscriptions__organization=organization,
+                    subscriptions__is_active=True,
+                )
+                .exists()
             )
-            .exists()
-        )
+            if not has_required:
+                return (False, None)
+
+        # Check population limits
+        population_limits = (self.config or {}).get("population_limits", {})
+        if not population_limits:
+            return (True, None)
+
+        # Check if operator can bypass population limits
+        if operator and (operator.config or {}).get(
+            "can_bypass_population_limits", False
+        ):
+            return (True, None)
+
+        # Check population limits based on organization type
+        commune_limit = population_limits.get("commune")
+        epci_limit = population_limits.get("epci")
+
+        # For communes: check if commune population < limit OR epci population < limit
+        if organization.type == "commune":
+            commune_pop = organization.population
+            epci_pop = organization.epci_population
+
+            # Block if both populations are null
+            if commune_pop is None and epci_pop is None:
+                return (False, "population_limit_exceeded")
+
+            # Check if at least one is below the limit
+            commune_ok = (
+                commune_limit is not None
+                and commune_pop is not None
+                and commune_pop < commune_limit
+            )
+            epci_ok = (
+                epci_limit is not None
+                and epci_pop is not None
+                and epci_pop < epci_limit
+            )
+
+            if commune_ok or epci_ok:
+                return (True, None)
+            return (False, "population_limit_exceeded")
+
+        # For EPCIs: check if epci population < limit
+        if organization.type == "epci":
+            if epci_limit is None:
+                return (True, None)
+
+            epci_pop = organization.population
+            if epci_pop is None:
+                return (False, "population_limit_exceeded")
```

Comment:
Ditto, maybe tell because that's not defined?

Reply:
should never happen